dashboard/Dockerfile

```diff
@@ -1,19 +1,22 @@
-FROM node:13.12.0-alpine
-
+FROM node:13.12.0-alpine as builder
 # set working directory
 WORKDIR /app
 
 # add `/app/node_modules/.bin` to $PATH
-ENV PATH /app/node_modules/.bin:$PATH
-ENV DIRECTORY dashboard
+ENV DIRECTORY ./dashboard
+
+# add app
+COPY ./$DIRECTORY/ ./
 
 # install app dependencies
-COPY $DIRECTORY/package.json $DIRECTORY/package-lock.json ./
+RUN npm install react-scripts -g
+RUN npm install typescript -g
 RUN npm install --silent
-#RUN npm install react-scripts -g
 
-# add app
-COPY ./$DIRECTORY ./
+RUN npm run build
+
+FROM nginx
+
+COPY --from=builder /app/build /usr/share/nginx/html
```

Comment:
I personally don't like this approach, the whole point of doing `npm start` is to have the benefits of debugging information and hot reloading, which you lose if you are running the build version.

Reply:
but in container you will not get hot reloading, code in container only changes when you build new image

Reply:
hmm, I do have hot reloading. We are mapping our local `dashboard` folder with the `app` folder on the container, so any local change automatically triggers a reloading on the dashboard app.